### ISSUE TITLE
Take and collect on ReadRDD.rdd throw ClassCastException

### ISSUE THREAD

**Issue by heuermh** [heuermh/biojava-adam#3]
```
scala> import org.biojava.nbio.adam.BiojavaAdamContext

scala> val biojavaContext = new BiojavaAdamContext(sc)

scala> val reads = biojavaContext.loadFastqReads("../adam/adam-core/src/test/resources/bqsr1-r1.fq")

scala> reads.rdd.take(4)
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [Lorg.bdgenomics.formats.avro.Read;
  ... 48 elided

scala> reads.rdd.collect.take(4)
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [Lorg.bdgenomics.formats.avro.Read;
  ... 48 elided

scala> reads.rdd.collect
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [Lorg.bdgenomics.formats.avro.Read;
  ... 48 elided

scala> reads.rdd.first
res1: org.bdgenomics.formats.avro.Read = {"name": "SRR062634.10022079/1", "description": "SRR062634.10022079/1", "alphabet": "DNA", "sequence": "AATTCAAAACCAGCCTGGCCAATATGGTGAAACCTCATCTCTACTAAAAATACAAAAATTAGCCAGGCATGGTGGTGCGTGCGTGTAGTCCCAGCTACTT", "length": 100, "qualityScores": "?-DDBEEB=EEEDDEDEEEA:D?5?E?CEBE5ED?D:AEDEDEDED-B,BC0AC,BB6@CDBDEC?BCBAA@5,=8CA-?A>?2:&048<BB5BE#####", "qualityScoreVariant": "FASTQ_SANGER", "attributes": {}}
```

**Comment by heuermh** [comment]
```
scala> val alignments = biojavaContext.loadAlignments("../adam/adam-core/src/test/resources/small.sam")
alignments: org.bdgenomics.adam.rdd.read.AlignmentRecordRDD = RDDBoundAlignmentRecordRDD with 2 reference sequences, 0 read groups, and 2 processing steps

scala> alignments.rdd.count()
res1: Long = 20

scala> alignments.rdd.take(4)
res2: Array[org.bdgenomics.formats.avro.AlignmentRecord] = Array({"readInFragment": 0, "contigName": "1", "start": 26472783, "oldPosition": null, "end": 26472858, "mapq": 60, "readName": "simread:1:26472783:false", "sequence": "GTATAAGAGCAGCCTTATTCCTATTTATAATCAGGGTGAAACACCTGTGCCAATGCCAAGACAGGGGTGCCAAGA", "qual": null, "cigar": "75M", "oldCigar": null, "basesTrimmedFromStart": 0, "basesTrimmedFromEnd": 0, "readPaired": false, "properPair": false, "readMapped": true, "mateMapped": false, "failedVendorQualityChecks": false, "duplicateRead": false, "readNegativeStrand": true, "mateNegativeStrand": false, "primaryAlignment": true, "secondaryAlignment": false, "supplementaryAlignment": false, "mismatchingPositions": null, "origQual": null, "attributes": "XS:i:0\tAS:i:75\tNM:i:0", "recordGroup...
scala> alignments.rdd.collect
res3: Array[org.bdgenomics.formats.avro.AlignmentRecord] = Array({"readInFragment": 0, "contigName": "1", "start": 26472783, "oldPosition": null, "end": 26472858, "mapq": 60, "readName": "simread:1:26472783:false", "sequence": "GTATAAGAGCAGCCTTATTCCTATTTATAATCAGGGTGAAACACCTGTGCCAATGCCAAGACAGGGGTGCCAAGA", "qual": null, "cigar": "75M", "oldCigar": null, "basesTrimmedFromStart": 0, "basesTrimmedFromEnd": 0, "readPaired": false, "properPair": false, "readMapped": true, "mateMapped": false, "failedVendorQualityChecks": false, "duplicateRead": false, "readNegativeStrand": true, "mateNegativeStrand": false, "primaryAlignment": true, "secondaryAlignment": false, "supplementaryAlignment": false, "mismatchingPositions": null, "origQual": null, "attributes": "XS:i:0\tAS:i:75\tNM:i:0", "recordGroup...
scala> alignments.rdd.collect.take(4)
res4: Array[org.bdgenomics.formats.avro.AlignmentRecord] = Array({"readInFragment": 0, "contigName": "1", "start": 26472783, "oldPosition": null, "end": 26472858, "mapq": 60, "readName": "simread:1:26472783:false", "sequence": "GTATAAGAGCAGCCTTATTCCTATTTATAATCAGGGTGAAACACCTGTGCCAATGCCAAGACAGGGGTGCCAAGA", "qual": null, "cigar": "75M", "oldCigar": null, "basesTrimmedFromStart": 0, "basesTrimmedFromEnd": 0, "readPaired": false, "properPair": false, "readMapped": true, "mateMapped": false, "failedVendorQualityChecks": false, "duplicateRead": false, "readNegativeStrand": true, "mateNegativeStrand": false, "primaryAlignment": true, "secondaryAlignment": false, "supplementaryAlignment": false, "mismatchingPositions": null, "origQual": null, "attributes": "XS:i:0\tAS:i:75\tNM:i:0", "recordGroup...
```

**Comment by heuermh** [comment]
A [Scala implementation of `BiojavaAdamContext`](https://github.com/heuermh/biojava-adam/blob/master/src/main/scala/org/biojava/nbio/adam/BiojavaAdamScalaContext.scala) may not have this issue
```
scala> import org.biojava.nbio.adam.BiojavaAdamScalaContext
import org.biojava.nbio.adam.BiojavaAdamScalaContext

scala> val biojavaScalaContext = BiojavaAdamScalaContext(sc)
biojavaScalaContext: org.biojava.nbio.adam.BiojavaAdamScalaContext = org.biojava.nbio.adam.BiojavaAdamScalaContext@74e4be21

scala> val reads = biojavaScalaContext.loadBiojavaFastqReads("../adam/adam-core/src/test/resources/bqsr1-r1.fq")
reads: org.bdgenomics.adam.rdd.read.ReadRDD = RDDBoundReadRDD with 0 reference sequences

scala> reads.rdd.count()
res0: Long = 488

scala> reads.rdd.first()
res1: org.bdgenomics.formats.avro.Read = {"name": "SRR062634.10022079/1", "description": "SRR062634.10022079/1", "alphabet": "DNA", "sequence": "AATTCAAAACCAGCCTGGCCAATATGGTGAAACCTCATCTCTACTAAAAATACAAAAATTAGCCAGGCATGGTGGTGCGTGCGTGTAGTCCCAGCTACTT", "length": 100, "qualityScores": "?-DDBEEB=EEEDDEDEEEA:D?5?E?CEBE5ED?D:AEDEDEDED-B,BC0AC,BB6@CDBDEC?BCBAA@5,=8CA-?A>?2:&048<BB5BE#####", "qualityScoreVariant": "FASTQ_SANGER", "attributes": {}}

scala> reads.rdd.take(4)
res2: Array[org.bdgenomics.formats.avro.Read] = Array({"name": "SRR062634.10022079/1", "description": "SRR062634.10022079/1", "alphabet": "DNA", "sequence": "AATTCAAAACCAGCCTGGCCAATATGGTGAAACCTCATCTCTACTAAAAATACAAAAATTAGCCAGGCATGGTGGTGCGTGCGTGTAGTCCCAGCTACTT", "length": 100, "qualityScores": "?-DDBEEB=EEEDDEDEEEA:D?5?E?CEBE5ED?D:AEDEDEDED-B,BC0AC,BB6@CDBDEC?BCBAA@5,=8CA-?A>?2:&048<BB5BE#####", "qualityScoreVariant": "FASTQ_SANGER", "attributes": {}}, {"name": "SRR062634.10072753/1", "description": "SRR062634.10072753/1", "alphabet": "DNA", "sequence": "TGTGACTAAGGACTGGTGTTCCTCTGGGGCCTCATGTCTACCTGGGCCCTGGGAGTGCACATGGAGCCAGATGTCTATAAAGGGCCTGAGTGTCCACTAG", "length": 100, "qualityScores": "FGDGGGGGGDGGGGGGFGGGEGGGGGFGGGGGGDGGGGGDFFDFCDFFFFEDDDEBBEDGEEEBABEEBBBBECCEEEEEEECEEC?C=CBCBCCCC@@?",...

scala> reads.rdd.collect
res3: Array[org.bdgenomics.formats.avro.Read] = Array({"name": "SRR062634.10022079/1", "description": "SRR062634.10022079/1", "alphabet": "DNA", "sequence": "AATTCAAAACCAGCCTGGCCAATATGGTGAAACCTCATCTCTACTAAAAATACAAAAATTAGCCAGGCATGGTGGTGCGTGCGTGTAGTCCCAGCTACTT", "length": 100, "qualityScores": "?-DDBEEB=EEEDDEDEEEA:D?5?E?CEBE5ED?D:AEDEDEDED-B,BC0AC,BB6@CDBDEC?BCBAA@5,=8CA-?A>?2:&048<BB5BE#####", "qualityScoreVariant": "FASTQ_SANGER", "attributes": {}}, {"name": "SRR062634.10072753/1", "description": "SRR062634.10072753/1", "alphabet": "DNA", "sequence": "TGTGACTAAGGACTGGTGTTCCTCTGGGGCCTCATGTCTACCTGGGCCCTGGGAGTGCACATGGAGCCAGATGTCTATAAAGGGCCTGAGTGTCCACTAG", "length": 100, "qualityScores": "FGDGGGGGGDGGGGGGFGGGEGGGGGFGGGGGGDGGGGGDFFDFCDFFFFEDDDEBBEDGEEEBABEEBBBBECCEEEEEEECEEC?C=CBCBCCCC@@?",...

scala> reads.rdd.collect.take(2)
res4: Array[org.bdgenomics.formats.avro.Read] = Array({"name": "SRR062634.10022079/1", "description": "SRR062634.10022079/1", "alphabet": "DNA", "sequence": "AATTCAAAACCAGCCTGGCCAATATGGTGAAACCTCATCTCTACTAAAAATACAAAAATTAGCCAGGCATGGTGGTGCGTGCGTGTAGTCCCAGCTACTT", "length": 100, "qualityScores": "?-DDBEEB=EEEDDEDEEEA:D?5?E?CEBE5ED?D:AEDEDEDED-B,BC0AC,BB6@CDBDEC?BCBAA@5,=8CA-?A>?2:&048<BB5BE#####", "qualityScoreVariant": "FASTQ_SANGER", "attributes": {}}, {"name": "SRR062634.10072753/1", "description": "SRR062634.10072753/1", "alphabet": "DNA", "sequence": "TGTGACTAAGGACTGGTGTTCCTCTGGGGCCTCATGTCTACCTGGGCCCTGGGAGTGCACATGGAGCCAGATGTCTATAAAGGGCCTGAGTGTCCACTAG", "length": 100, "qualityScores": "FGDGGGGGGDGGGGGGFGGGEGGGGGFGGGGGGDGGGGGDFFDFCDFFFFEDDDEBBEDGEEEBABEEBBBBECCEEEEEEECEEC?C=CBCBCCCC@@?",...
```

**Comment by heuermh** [comment]
Fixed by commit 19dedce97deebc324c2fa78221b82fbfa30e1904
